aws/sdk/examples/dynamodb/src/bin/crud.rs

```diff
@@ -0,0 +1,457 @@
+    let last_av = AttributeValue::S(item.last_name);
+
+    match client
+        .put_item()
+        .table_name(item.table)
+        .item(item.key, user_av)
+        .item("account_type", type_av)
+        .item("age", age_av)
+        .item("first_name", first_av)
+        .item("last_name", last_av)
+        .send()
+        .await
+    {
+        Ok(_) => println!(),
+        Err(e) => {
+            println!("Got an error adding item to table:");
+            println!("{}", e);
+            process::exit(1);
+        }
+    }
+}
+
+/// Query the table for an item matching the input values.
+async fn query(client: &dynamodb::Client, item: Item) {
+    let value = &item.value;
+    let key = &item.key;
+    let user_av = AttributeValue::S(value.to_string());
+    let userav = AttributeValue::S(value.to_string());
+    let type_av = AttributeValue::S(item.utype);
+    let age_av = AttributeValue::S(item.age);
+    let first_av = AttributeValue::S(item.first_name);
+    let last_av = AttributeValue::S(item.last_name);
+    let mut found_match = true;
+    let mut cond = HashMap::new();
+    cond.insert("#key".to_string(), key.to_string());
+    let mut expr = HashMap::new();
+    expr.insert(":value".to_string(), user_av);
+    match client
+        .query()
+        .table_name(item.table)
+        .set_key_condition_expression(Some("#key = :value".to_string()))
+        .set_expression_attribute_names(Some(cond))
+        .set_expression_attribute_values(Some(expr))
+        .select(Select::AllAttributes)
+        .send()
+        .await
+    {
+        Ok(resp) => {
+            let items = resp.items.unwrap_or_default();
+            for item in items {
+                // Do key values match?
+                match item.get(&String::from(key)) {
+                    None => found_match = false,
+                    Some(v) => {
+                        if v != &userav {
+                            found_match = false;
+                        }
+                    }
+                }
+
+                // Do age values match?
+                match item.get(&String::from("age")) {
+                    None => found_match = false,
+                    Some(v) => {
+                        if v != &age_av {
+                            found_match = false;
+                        }
+                    }
+                }
+
+                // Do first name values match?
+                match item.get(&String::from("first_name")) {
+                    None => found_match = false,
+                    Some(v) => {
+                        if v != &first_av {
+                            found_match = false;
+                        }
+                    }
+                }
+
+                // Do last name values match?
+                match item.get(&String::from("last_name")) {
+                    None => found_match = false,
+                    Some(v) => {
+                        if v != &last_av {
```

Comment:
is this still needed? isn't this handled by the query clause?

Reply:
Good point. I'll remove all the checks.